sklearn/model_selection/tests/test_split.py

```diff
@@ -702,6 +716,66 @@ def test_group_shuffle_split():
                             round((1.0 - test_size) * len(l_unique))) <= 1)
 
 
+def test_leave_one_p_group_out():
+    logo = LeaveOneGroupOut()
+    lpgo_1 = LeavePGroupsOut(n_groups=1)
+    lpgo_2 = LeavePGroupsOut(n_groups=2)
+    lpgo_3 = LeavePGroupsOut(n_groups=3)
+
+    # Make sure the repr works
+    assert_equal(repr(logo), 'LeaveOneGroupOut()')
+    assert_equal(repr(lpgo_1), 'LeavePGroupsOut(n_groups=1)')
+    assert_equal(repr(lpgo_2), 'LeavePGroupsOut(n_groups=2)')
+    assert_equal(repr(lpgo_3), 'LeavePGroupsOut(n_groups=3)')
+
+    for j, (cv, p_groups_out) in enumerate(((logo, 1), (lpgo_1, 1),
+                                            (lpgo_2, 2))):
+        groups = (np.array([1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 3, 3]),
+                  np.array([0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3]),
+                  np.array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2]),
+                  np.array([1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4]),
+                  [1, 1, 1, 1, 2, 2, 2, 3, 3, 3, 3, 3],
+                  ['1', '1', '1', '1', '2', '2', '2', '3', '3', '3', '3', '3'])
+
+        all_n_splits = np.array([[3, 3, 3],
```

Comment:
> It's just scipy.misc.comb(len(np.unique(groups_i)), p_groups_out) right

That is the implementation in `_split.py`. I thought it would be better to compare it against hand calculated values?